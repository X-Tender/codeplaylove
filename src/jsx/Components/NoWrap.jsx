import React from 'react';

const NoWrap = ({ children }) => <span className="no-wrap">{children}</span>;

export default NoWrap;
