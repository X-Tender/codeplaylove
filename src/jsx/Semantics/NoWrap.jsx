import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	children: PropTypes.node.isRequired,
};

const NoWrap = ({ children }) => <span className="no-wrap">{children}</span>;

NoWrap.propTypes = propTypes;

export default NoWrap;
