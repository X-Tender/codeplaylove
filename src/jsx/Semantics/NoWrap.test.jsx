import React from 'react';
import { shallow } from 'enzyme';
import NoWrap from './NoWrap';

describe('NoWrap', () => {
	const children = <p>Hello world</p>;
	const comp = shallow(<NoWrap>{children}</NoWrap>);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders props correctly', () => {
		expect(comp.find('span.no-wrap').contains(children)).toBe(true);
	});
});
