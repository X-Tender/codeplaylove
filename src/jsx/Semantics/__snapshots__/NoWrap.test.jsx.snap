// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoWrap renders correctly 1`] = `
<span
  className="no-wrap"
>
  <p>
    Hello world
  </p>
</span>
`;
